addons/website/static/src/snippets/s_alert/options.js

```diff
@@ -0,0 +1,21 @@
+odoo.define('website.s_alert_options', function (require) {
+'use strict';
+
+var options = require('web_editor.snippets.options');
+
+options.registry.Alert = options.Class.extend({
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    selectType: function (previewMode, widgetValue, params) {
+        const $snippet = this.$target.parent('.s_alert');
+        $snippet.removeClass('alert-delta'); // TODO (review) - migration/compatibility
+        $snippet.removeClass(params.possibleValues.join(' '));
+        $snippet.addClass(widgetValue);
```

Comment:
Mmmh... all of this is kinda strange :/ You have to find back the widget because you used an apply-to, you must reimplement selectClass for the alert- stuff etc :/ and this cannot even work since you don't know the colors to remove (your TODO).

Note: this also revealed a bug I will fix in stable versions as well (the info/success/... colors cannot be removed).

Maybe the best is to do a hack for now: search for the colorpicker widget in your option DOM and click on the appropriate color by hand when selecting a specific type ? (add a comment explaining)

Reply:
I am not sure to fully understand the issue but this is working as I expect it to?
When you select one of the available preset type (danger, warning..) it will use the related color.
If you want another one, using the colorpicker will prevail on the preset type, then you can switch type without it changing the color as you selected a specific color ?
So specific color > layout color, but you can still change layout to switch icon etc.

> this cannot even work since you don't know the colors to remove (your TODO).

No, the TODO is just for compatibility with older snippet which have the `alert-delta` class that I removed, just that line makes it compatible (otherwise it would not work and we would need a new snippet version)

And I don't think it is weird to find the parent? I need to toggle classes on 2 different DOM elements with the same option, rather use the native select-class for one and a method for the other, no?

Reply:
Personally, when you select the type, I would clearly want the color to change back again (and that's why type should be first). The status color do not even appears in the colorpicker :/ Personnaly, what I would do as an user is:
0. drop the alert snippet
1. add the alert text
2. select the alert type
3. try another color just for fun
4. trying to revert back to the correct type color... which I can't without doing all over again.

No? @sbuhl ?

Reply:
You just need to reset the colorpicker with the reset button

I'd type my text, select the color I am fine with (probably a light blue/gray) and then change type to see the icons but I really don't think there is a 'normal' flow for this one..

That'd as much looks buggy to me if it would change the color I just chose when I change the type of the alert (which is not a color but multiple set up, color and icon in this case).

I don't care tbh or it will never end :zzz:

Reply:
Well actually I do care, I think losing the color I chose and I wont be able to find back just because I wanted to select another icon is not cool.
How I implemented it won't lose anything, it's just the selected color that prevail but you can unset it through the colorpicker if you don't want it anymore.

Reply:
Ah ah I prefer when you care :p I think this is important if we want to make the UI perfect. I suppose it is a matter of opinion but I don't agree with you here. It is the same with other options, you set a color, then set a background (below option), if you want to change the color again, your background is removed. Otherwise you do not have any apparent effect for the color, which is the main option.

In your case, why would you be ok to lose the icon when you change the type but not ok to lose the color (to match the type). Also you lose it only if you click on a type, you can also not choose anything after previewing them.

Anyways, I'll @sbuhl decide :+1:

Reply:
Despite the fact that the behavior would change if using colorpicker theme color or using other colors, I stay on my position where I'd not want my so-hardly-chosen color to be erased (even in preview mode) when changing type (even if DOM not 100% bootstrap compliant in some cases with theme color). Indeed, I change the **type/display/layout**, not the color, which is only a part of the type/display/layout.

But as long discussed today, I don't think we will come to a consensus anytime soon. And since it has no value whatsoever to implement one or the other solution (the snippet value would probably increase/decrease by 0.001% depending of your position), let's do what you propose so we move forward:

Changing **layout** will erase the chosen color.

Note that I left a TODO in it while your fix/imp is not merged yet. Without it it won't work except for 2 colors.